pytube/cli.py

```diff
@@ -426,7 +432,10 @@ def download_by_resolution(
         Target directory for download
     """
     # TODO(nficano): allow dash itags to be selected
-    stream = youtube.streams.get_by_resolution(resolution)
+    if resolution == "best":
```

Comment:
I think we can do this a better way, especially since this will cause the error to read funny if a stream isn't found ("Could not find a stream with resolution: best").

Reply:
Notably, there's a conflict with the `--ffmpeg` [argument](https://github.com/pytube/pytube/blob/master/pytube/cli.py#L152) which downloads and merges the highest quality video and audio streams.

Reply:
> I think we can do this a better way, especially since this will cause the error to read funny if a stream isn't found ("Could not find a stream with resolution: best").

will such an Error really occur since  youtube.streams.get_highest_resolution() should always provide a video stream (what ever the best stream is)  ? or is there a case thinkable, where you youtube will not provide a video stream with some resolution ?

Reply:
> Notably, there's a conflict with the `--ffmpeg` [argument](https://github.com/pytube/pytube/blob/master/pytube/cli.py#L152) which downloads and merges the highest quality video and audio streams.

You mean that _ffmepg_downloader() would be also triggered by this ? then the command line arguments are not well designed ...

Reply:
You're right, it *shouldn't* cause that error to happen now that I think about it. When I wrote this comment originally, I meant to follow up on it a bit. I think offering the "best" resolution by default isn't a great idea, which I talk about in more detail [here](https://github.com/pytube/pytube/issues/901#issuecomment-753535009). 

We should probably spend a bit more time with this PR before we merge it to iron this out. In a lot of cases, the "best" resolution provided by get_highest_resolution() isn't actually the best possible because of adaptive streams. The `-f` flag will provide the highest quality in most cases, but introduces an external dependency, `ffmpeg`.

Reply:
Regarding the "best" resolution, I made some changes that avoid this issue completely. youtube.streams.get_highest_resolution() should throw an pytube.exceptions.VideoUnavailable error to be completely safe.

Switching to option 2 (output some help if no parameter is provided is also ok - but not so intuitive, since people tend to forget options) - see my comment in #901.